src/components/visualizations/chart/chartOptionsBuilder.ts

```diff
@@ -1334,17 +1337,15 @@ function createYAxisItem(measuresInAxis: any[], opposite = false) {
 }
 
 function assignYAxes(series: any, yAxes: IAxis[]) {
-    series.forEach((seriesItem: any, index: number) => {
+    return series.reduce((result: any, _: any, index: number) => {
         const yAxisIndex = yAxes.findIndex((axis: IAxis) => {
             return includes(get(axis, 'seriesIndices', []), index);
         });
-
         // for case viewBy and stackBy have one attribute, and one measure is sliced to multiple series
         // then 'yAxis' in other series should follow the first one
-        seriesItem.yAxis = yAxisIndex !== -1 ? yAxisIndex : series[0].yAxis;
-    });
-
-    return series;
+        result[index].yAxis = yAxisIndex !== -1 ? yAxisIndex : result[0].yAxis;
+        return result;
+    }, [...series]);
```

Comment:
I would say this is solving mutation issue only partially... basically you created a new array of seriesItems but seriesItems it self they are the same objects between two arrays... :/ Am I  right?

consider something like this please:
```
return series.reduce(result: any, item:any, index: number) => {
	...

	const seriesItem: {
		...item,
		yAxis: yAxisIndex !== -1 ? yAxisIndex : result[0].yAxis
	};

	result.push(seriesItem);
	return result;
}, []);
```

Reply:
it's done :)